cheatsheets/Protect_FileUpload_Against_Malicious_File.md

```diff
@@ -1,597 +1,138 @@
-                                break;
-                            }
-                            case "DCX": {
-                                imageParser = new DcxImageParser();
-                                break;
-                            }
-                            case "BMP": {
-                                imageParser = new BmpImageParser();
-                                break;
-                            }
-                            case "GIF": {
-                                imageParser = new GifImageParser();
-                                break;
-                            }
-                            case "PNG": {
-                                imageParser = new PngImageParser();
-                                break;
-                            }
-                            case "WBMP": {
-                                imageParser = new WbmpImageParser();
-                                break;
-                            }
-                            case "XBM": {
-                                imageParser = new XbmImageParser();
-                                break;
-                            }
-                            case "XPM": {
-                                imageParser = new XpmImageParser();
-                                break;
-                            }
-                            default: {
-                                throw new IOException("Format of the original image is not" + 
-                                                      " supported for write operation !");
-                            }
-
-                        }
-                        imageParser.writeImage(sanitizedImage, fos, new HashMap<>());
-                    }
-
-                }
-
-                // Set state flag
-                safeState = true;
-            }
-        } catch (Exception e) {
-            safeState = false;
-            LOG.warn("Error during Image file processing !", e);
-        }
-
-        return safeState;
-    }
-}
+```python
+# This code snippet only serves as an example, and should not be used
+blacklist = ["php", "php5"]  # can be replaced with a regex pattern, but that would be even weaker, such as \.php$
+for bad in blacklist:
+    if "file.php".endswith(bad):
+        // raise exception
 ```
 
-# Sources of the prototype
+### User Permissions
+
+Before any file upload service is accessed, proper validation should occur on two levels for the user uploading a file:
+
+- Authentication level
+  - The user should be a registered user, or an identifiable user, in order to set restrictions and limitations for their upload capabilities
+- Authorization level
+  - Make sure that the user's accesses allow them to upload a file, or multiple files for administrative cases.
+
+### Content-Type Validation
+
+Other than defining the extension of the uploaded file, its MIME-type can be checked for a quick protection against simple file upload attacks.
```

Comment:
Not really clear how you'd `check` it - I'm assuming matching the content type to the file type or whitelist?

Needs strong emphasis that this is trivially spoofed and should not be relied upon.

Reply:
There are libraries like Apache Tika that can sniff the file content and compare to the extension

Reply:
@rbsec I have added a warning in this section, and in the introduction. I won't be adding anything to this. Content Type validation can be conducted fairly easily by almost all web frameworks.